From 3f88ddb71ba49d343a5db1304c296e78ddeb2575 Mon Sep 17 00:00:00 2001
From: Koan-Sin Tan <koansin.tan@gmail.com>
Date: Wed, 10 Oct 2018 02:34:02 +0000
Subject: [PATCH] [aarch64] make aws sdk work on aarch64

`bazel build //tensorflow/tools/pip_package:build_pip_package'
requires AWS SDK by default. but platform part was not built
on aarch64
---
 tensorflow/BUILD            | 6 ++++++
 third_party/aws/BUILD.bazel | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 9b62a504525d..8486922e00b0 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -163,6 +163,12 @@ config_setting(
     visibility = ["//visibility:public"],
 )
 
+config_setting(
+    name = "linux_aarch64",
+    values = {"cpu": "aarch64"},
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "linux_x86_64",
     values = {"cpu": "k8"},
diff --git a/third_party/aws/BUILD.bazel b/third_party/aws/BUILD.bazel
index 5426f79e4650..66baa8fdf3b7 100644
--- a/third_party/aws/BUILD.bazel
+++ b/third_party/aws/BUILD.bazel
@@ -12,6 +12,9 @@ load("@org_tensorflow//third_party:common.bzl", "template_rule")
 cc_library(
     name = "aws",
     srcs = select({
+        "@org_tensorflow//tensorflow:linux_aarch64": glob([
+            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
+        ]),
         "@org_tensorflow//tensorflow:linux_x86_64": glob([
             "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
         ]),
